include/seqan3/std/iterator

```diff
@@ -55,6 +55,73 @@
 namespace std
 {
 
+/*!\defgroup iterator_associated_types Associated types
+ * \ingroup iterator
+ * \brief Iterator requirements
+ */
```

Comment:
you are not defining a group here, but a (sub-)module, is that intended?

Reply:
In general we need to discuss if you shouldn't maybe remove all doc in the std module and finally get the plugin working that redirects to cppreference.com

Reply:
Yes, that would be good. The added subgroup follows the sections of the standard draft, in case other assoicated types will be added as well.

Reply:
Yes, but what I wanted to say was: We have a very clear structure in our documentation:
Modules and sub-modules correspond to folders and sub-folders.

The std-module is breaking this.

Reply:
Yes, I am seeing your concern. But I can't group them otherwise, can I?

Reply:
Maybe `\section`?

Reply:
Ok, I couldn't find anything usful. I really don't care. I can revert this to make it plain in the iterator and we can change it later, if we decide for it.